src/audio/base_fw.c

```diff
@@ -323,6 +325,71 @@ static uint32_t basefw_get_ext_system_time(uint32_t *data_offset, char *data)
 	return IPC4_UNAVAILABLE;
 }
 
+static int basefw_register_kcps(bool first_block,
+				bool last_block,
+				uint32_t data_offset_or_size,
+				const char *data)
+{
+	if (!(first_block && last_block))
+		return IPC4_INVALID_REQUEST;
+
+	/* value of kcps to request on core 0. Can be negative */
+	if (core_kcps_adjust(0, *(int32_t *)data))
+		return IPC4_FAILURE;
+
+	return IPC4_SUCCESS;
```

Comment:
This needs to be POSIX and do the IPC4 translation in the IPC layer.

Reply:
I see, in base_fw.c they are mixed up. I suppose that's just lack of cleanup?